lang/syntax.md

```diff
@@ -14,11 +14,14 @@ pub struct Program {
     pub functions: Map<FnName, Function>,
     /// The function where execution starts.
     pub start: FnName,
+    /// Associate each global name with the associated global.
+    pub globals: Map<GlobalName, Global>,
```

Comment:
There needs to be some wf checking for this. At least all Globalname occurring everywhere must actually be present in this list. And probably in a Global, we want to make sure that the relocations are in-bounds? If `bytes` has length 4 then a relocation at offset 2 doesn't work out... that might even lead to panics in `Machine::new`? And finally, we might want to sanity-check the offset on the relocations and const pointers: maybe we want to require it to be in-bounds of the global it points to? That should then ensure that `Machine::new` is never UB due to pointer arithmetic overflow.